### DISCUSSION TITLE
Reducing GPU memory consumption for (TTS) training

### DISCUSSION THREAD

**Discussion by padmalcom** [speechbrain/speechbrain#1625] [General]
Hi, I'm training a German TTS tacotron2 model on a custom dataset. I'm limited to 24GB GPU memory and the training process runs in an "CUDA out of memory" error. So what I do is to reduce the batch size (from 64 I'm down to 4 now).

Do I have any other configuration parameters that I can chance to fit my training batches in the GPU memory?

Thanks in advance and thank your for your great framework.

**Comment by Adel-Moumen** [comment]
Hello @BenoitWang, could you please take a look? Thanks! :-)

**Comment by BenoitWang** [comment]
HI @padmalcom, 
You can add a gradient accumulation when reducing the batchsize for a more stable convergence. Or check the lengths of audio maybe they are much longer than 10 seconds. And finally make sure to check the audio parameters (sample rate, etc.) and to use a German text pipeline.
Hope this can help.

**Comment by padmalcom** [comment]
Hi @BenoitWang,
thank you, I reduced the sample rate to 16000 and changed the english_cleaners to a basic_cleaners as I've found is recommended for German text. Thank you for pointing out the audio length which definitely will be the issue here. I wrote a tool to generate training data from audio books and looking for the longest sentences I found that there are some wavs with a length of > 60 seconds. I'll remove them and restart the training.
